.github/workflows/release.yml

```diff
@@ -0,0 +1,57 @@
+name: Release
+
+permissions:
+  contents: write
+
+on:
+  push:
+    tags:
+      - mountpoint-s3-[0-9]+.*
+      - mountpoint-s3-[a-z]+-[0-9]+.*
+
+jobs:
+  create-release:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: taiki-e/create-gh-release-action@v1
+      with:
+        prefix: mountpoint-s3(-[a-z]+)?
+        draft: true
+        title: "$prefix $version"
+        token: ${{ secrets.GITHUB_TOKEN }}
+
+  upload-assets:
+    name: Build and release on target (${{ matrix.target }})
+    runs-on: ${{ matrix.os }}
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        target: [x86_64-unknown-linux-gnu]
+    
+    steps:
+    - name: Checkout source code
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+    - name: Set up stable Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        override: true
+    - name: Install operating system dependencies
+      uses: ./.github/actions/install-dependencies
+      with:
+        fuseVersion: 2 
+    - name: Release mount-s3 binary
+      uses: taiki-e/upload-rust-binary-action@v1
+      with:
+        bin: mount-s3
+        token: ${{ secrets.GITHUB_TOKEN }}
+        asset: LICENSE,README.md
+        target:  ${{ matrix.target }}
+        archive: $bin-$tag-$target
```

Comment:
probably don't need `$bin` here since the `tag` already starts with `mountpoint-s3-`